### PR TITLE
HOLD Software: enumerate additional service-accounts that use backends

### PR DESCRIPTION
**Pull request by rob-1126**
One of our VIP customers wanted more information about the service-accounts and a listing of all those that might need xcom access. This updates our previously incomplete list and adds a bit more detail to make this VIP happy.

### CODE REVIEW

**Comment by lzdanski** [comment]
@rob-1126 - could you possibly update your branch with a more recent version of `main`? I'm seeing a bunch of build errors and I think it might be because we made quite a few infrastructure changes recently.

File: software/integrate-iam.md [hunk 1]
```diff
@@ -326,3 +327,16 @@ astronomer:
...
+| `<release name>-scheduler`         | Customer-provided dag-code is regularly interpreted to determine what tasks are part of dags, and that code may incorporate references to values stored in secret storage or XCom.                  |
+| `<release name>-worker`            | Customer-provided dag-code is interpreted at task-run time and may incorporate references to values stored in secret storage or XCom.             |
+| `<release name>-triggerer`         | Customer-provided dag-code is interpreted at task-run time and may incorporate references to values stored in secret storage or XCom.            |
+| `<release name>-webserver`         | Provides users a mechanism to view and set secrets and to view XCom entries.|
+| `<release name>-cleanup`           |  Accesses task instances and other data which may include serialized references to values stored in secret-backends or xcom-backends.       |
+| `<release name>-migrate-database-job` | Analyzes serialized dag models which may include serialized references to values stored in secret-backends or xcom-backends. |
```
Comment (lzdanski) [inline software/integrate-iam.md:342]: Devastated and appalled to learn the plural of "Secrets backend" is "secrets backends"

In this, is the "customer" the Astro user? Would it be more accurate to say "your DAG code"?

```suggestion
| `<release name>-scheduler`         | Customer-provided DAG code is regularly interpreted to determine what tasks are part of DAGs, and that code might incorporate references to values stored in secret storage or XCom.                  |
| `<release name>-worker`            | Customer-provided DAG code is interpreted at task-run time and might incorporate references to values stored in secret storage or XCom.             |
| `<release name>-triggerer`         | Customer-provided DAG code is interpreted at task run time and might incorporate references to values stored in secrets storage or XCom.            |
| `<release name>-webserver`         | Provides users a mechanism to view and set secrets and to view XCom entries.|
| `<release name>-cleanup`           |  Accesses task instances and other data that might include serialized references to values stored in secrets backends or XCOM backends.       |
| `<release name>-migrate-database-job` | Analyzes serialized DAG models that might include serialized references to values stored in secrets backends or XCOM backends. |
```

File: software/integrate-iam.md [hunk 2]
```diff
@@ -326,3 +327,16 @@ astronomer:
     ```
 
     If Workload Identity is working, you should see a list of credentialed accounts related to your GCP service account.
+
+## Granting Access to Airflow Components Using XCom and Secret Backends {sas-that-use-backends}
```
Comment (lzdanski) [inline software/integrate-iam.md:331]:
```suggestion
## Granting Access to Airflow components using XCom and secrets backends {sas-that-use-backends}
```

File: software/integrate-iam.md [hunk 3]
```diff
@@ -326,3 +327,16 @@ astronomer:
...
+Astronomer Software creates a set of service-accounts for each Airflow instance it manages.
+The following roles can require access to Airflow Xcom-backends or secret-backends to function:
```
Comment (lzdanski) [inline software/integrate-iam.md:333]:
```suggestion
Astronomer Software creates a set of service-accounts for each Airflow instance it manages. The following roles can require access to Airflow Xcom backends or secrets backends to function:
```

**Review by lzdanski — DISMISSED** [review]
question about who the table is referring to with "customer-provided"

**Comment by lzdanski** [comment]
Also can you try rebasing/ updating from main? We debugged a whole bunch of build errors earlier this week and I think it might still be causing some failures here if the branch hasn't been updated.

**Comment by rob-1126** [comment]
moving to https://github.com/astronomer/docs/pull/3940

**Pull request closed without merging**